src/formatting/lists.rs

```diff
@@ -325,7 +329,11 @@ where
                 }
             }
             DefinitiveListTactic::Vertical
-                if !first && !inner_item.is_empty() && !result.is_empty() =>
+                if !first
+                    && (!inner_item.is_empty()
+                        || item.post_comment.is_some()
+                        || item.pre_comment.is_some())
+                    && !result.is_empty() =>
```

Comment:
This guard is starting to get pretty gnarly. Could you refactor it a bit to improve readability? We don't have any other subsequent `Vertical` arms, so I suppose some of these checks could be moved into the body (since it's a noop if conditions aren't met).

For example i think to keep this in the guard we'd want to potentially shift this to some locals like

```rust
let item_has_comments = item.post_comment.is_some() || item.pre_comment.is_some();
// or
let non_empty_or_comments = !inner_item.is_empty() || item.post_comment.is_some() || item.pre_comment.is_some();
```
Otherwise I think it'd probably be much more readable to reduce the guard conditions and use these checks in the body to determine whether to push.

Also could we just do a single `result.push(newline_str)` now since that's declared above (vs. `result.push('\n'); result.push(indent_str);`)